integrations/octopus/client.py

```diff
@@ -69,15 +71,29 @@ async def get_paginated_resources(
             params["skip"] += PAGE_SIZE
             page += 1
 
+    async def get_paginated_resources(
+        self,
+        kind: str,
+        params: Optional[dict[str, Any]] = None,
+        path_parameter: Optional[str] = None,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        endpoint = f"{path_parameter}/{kind}s" if path_parameter else f"{kind}s"
+        async for resource_batch in self._get_paginated_resources(
+            endpoint, params, kind
+        ):
+            yield resource_batch
+
     async def get_single_resource(
-        self, resource_kind: str, resource_id: str
+        self, resource_kind: str, resource_id: str, space_id: str
```

Comment:
what happens if you send it an empty string? as you do when you get event from a real time webhook

Reply:
It will return 404, but the code has been slightly modified so it will always have a space_id when making this call. All events has the space_id present in the payload